clippy_lints/src/manual_non_exhaustive.rs

```diff
@@ -55,10 +58,28 @@ declare_clippy_lint! {
     "manual implementations of the non-exhaustive pattern can be simplified using #[non_exhaustive]"
 }
 
-declare_lint_pass!(ManualNonExhaustive => [MANUAL_NON_EXHAUSTIVE]);
+#[derive(Clone)]
+pub struct ManualNonExhaustive {
+    msrv_stack: LimitStack<String>,
+}
+
+impl ManualNonExhaustive {
+    #[must_use]
+    pub fn new(msrv: String) -> Self {
+        Self {
+            msrv_stack: LimitStack::new(msrv),
+        }
+    }
+}
+
+impl_lint_pass!(ManualNonExhaustive => [MANUAL_NON_EXHAUSTIVE]);
 
 impl EarlyLintPass for ManualNonExhaustive {
     fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
+        if Version::parse(&self.msrv_stack.limit()) < Version::parse(MANUAL_NON_EXHAUSTIVE_MSRV) {
+            return;
```

Comment:
```suggestion
            return
```
Would this nitpick be caught by `rustfmt`?